dfget/core/api/supernode_api.go

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 1999-2018 Alibaba Group.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/alibaba/Dragonfly/dfget/types"
+	"github.com/alibaba/Dragonfly/dfget/util"
+)
+
+/* the url paths of supernode APIs*/
+const (
+	PeerRegisterPath      = "/peer/registry"
+	PeerPullPieceTaskPath = "/peer/task"
+	PeerReportPiecePath   = "/peer/piece/suc"
+	PeerServiceDownPath   = "/peer/service/down"
+)
+
+// NewSupernodeAPI creates a new instance of SupernodeAPI with default value.
+func NewSupernodeAPI() SupernodeAPI {
```

Comment:
It might be better to allow initialization with parameters.

Reply:
I think that it could write another function to initialize with customized parameters, and this function just creates an api with default value.

Reply:
That works, too